Tracing tests fail on RHEL 10 due to missing `ps`

### Generated by Generative AI

nope

### Operating System:

RHEL 10

### ROS version or commit hash:

rolling

### RMW implementation (if applicable):

_No response_

### RMW Configuration (if applicable):

_No response_

### Client library (if applicable):

_No response_

### 'ros2 doctor --report' output

<details><summary>ros2 doctor --report</summary>

```console
<COPY OUTPUT HERE>
```
</details>


### Steps to reproduce issue

Run tracing tests on RHEL 10, specifically `test_tracetools_launch` and `test_tracetools` tests.

### Expected behavior

Tests pass.

### Actual behavior

Tests fail during call to `lttng_impl.is_session_daemon_unreachable()` due to subprocess call to `ps`: https://github.com/ros2/ros2_tracing/blob/0e868cc5ae89865aba75d305c894b65821efd2b5/tracetools_trace/tracetools_trace/tools/lttng_impl.py#L165

See https://ci.ros2.org/job/test_ci_linux-rhel/64/consoleFull:

<details><summary>test_tracetools_launch failure stacktrace</summary>

```console
22:44:51 [DEBUG] [launch]: An exception was raised in an async action/event
22:44:51 [DEBUG] [launch]: Traceback (most recent call last):
22:44:51   File "/home/jenkins-agent/workspace/test_ci_linux-rhel/ws/install/launch/lib/python3.12/site-packages/launch/launch_service.py", line 354, in run_async
22:44:51     raise exception_to_raise
22:44:51   File "/home/jenkins-agent/workspace/test_ci_linux-rhel/ws/install/launch/lib/python3.12/site-packages/launch/launch_service.py", line 241, in _process_one_event
22:44:51     await self.__process_event(next_event)
22:44:51   File "/home/jenkins-agent/workspace/test_ci_linux-rhel/ws/install/launch/lib/python3.12/site-packages/launch/launch_service.py", line 261, in __process_event
22:44:51     visit_all_entities_and_collect_futures(entity, self.__context))
22:44:51     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
22:44:51   File "/home/jenkins-agent/workspace/test_ci_linux-rhel/ws/install/launch/lib/python3.12/site-packages/launch/utilities/visit_all_entities_and_collect_futures_impl.py", line 45, in visit_all_entities_and_collect_futures
22:44:51     futures_to_return += visit_all_entities_and_collect_futures(sub_entity, context)
22:44:51                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
22:44:51   File "/home/jenkins-agent/workspace/test_ci_linux-rhel/ws/install/launch/lib/python3.12/site-packages/launch/utilities/visit_all_entities_and_collect_futures_impl.py", line 45, in visit_all_entities_and_collect_futures
22:44:51     futures_to_return += visit_all_entities_and_collect_futures(sub_entity, context)
22:44:51                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
22:44:51   File "/home/jenkins-agent/workspace/test_ci_linux-rhel/ws/install/launch/lib/python3.12/site-packages/launch/utilities/visit_all_entities_and_collect_futures_impl.py", line 38, in visit_all_entities_and_collect_futures
22:44:51     sub_entities = entity.visit(context)
22:44:51                    ^^^^^^^^^^^^^^^^^^^^^
22:44:51   File "/home/jenkins-agent/workspace/test_ci_linux-rhel/ws/install/launch/lib/python3.12/site-packages/launch/action.py", line 116, in visit
22:44:51     return self.execute(context)
22:44:51            ^^^^^^^^^^^^^^^^^^^^^
22:44:51   File "/home/jenkins-agent/workspace/test_ci_linux-rhel/ws/install/tracetools_launch/lib/python3.12/site-packages/tracetools_launch/action.py", line 512, in execute
22:44:51     if not setup():
22:44:51            ^^^^^^^
22:44:51   File "/home/jenkins-agent/workspace/test_ci_linux-rhel/ws/install/tracetools_launch/lib/python3.12/site-packages/tracetools_launch/action.py", line 474, in setup
22:44:51     self._trace_directory = lttng.lttng_init(
22:44:51                             ^^^^^^^^^^^^^^^^^
22:44:51   File "/home/jenkins-agent/workspace/test_ci_linux-rhel/ws/install/tracetools_trace/lib/python3.12/site-packages/tracetools_trace/tools/lttng.py", line 56, in lttng_init
22:44:51     trace_directory = _lttng.setup(**kwargs)
22:44:51                       ^^^^^^^^^^^^^^^^^^^^^^
22:44:51   File "/home/jenkins-agent/workspace/test_ci_linux-rhel/ws/install/tracetools_trace/lib/python3.12/site-packages/tracetools_trace/tools/lttng_impl.py", line 274, in setup
22:44:51     if is_session_daemon_unreachable():
22:44:51        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
22:44:51   File "/home/jenkins-agent/workspace/test_ci_linux-rhel/ws/install/tracetools_trace/lib/python3.12/site-packages/tracetools_trace/tools/lttng_impl.py", line 164, in is_session_daemon_unreachable
22:44:51     process = subprocess.run(
22:44:51               ^^^^^^^^^^^^^^^
22:44:51   File "/usr/lib64/python3.12/subprocess.py", line 548, in run
22:44:51     with Popen(*popenargs, **kwargs) as process:
22:44:51          ^^^^^^^^^^^^^^^^^^^^^^^^^^^
22:44:51   File "/usr/lib64/python3.12/subprocess.py", line 1026, in __init__
22:44:51     self._execute_child(args, executable, preexec_fn, close_fds,
22:44:51   File "/usr/lib64/python3.12/subprocess.py", line 1955, in _execute_child
22:44:51     raise child_exception_type(errno_num, err_msg, err_filename)
22:44:51 FileNotFoundError: [Errno 2] No such file or directory: 'ps'
22:44:51 
22:44:51 [ERROR] [launch]: Caught exception in launch (see debug for traceback): [Errno 2] No such file or directory: 'ps'
```
</details>

<details><summary>test_tracetools error</summary>

```console
22:41:18 32: =================================== FAILURES ===================================
22:41:18 32: __________________________ TestSubscription.test_all ___________________________
22:41:18 32: ../../../../install/tracetools_test/lib/python3.12/site-packages/tracetools_test/case.py:110: in setUp
22:41:18 32:     self.assertEqual(exit_code, 0)
22:41:18 32: E   AssertionError: 1 != 0
22:41:18 32: ----------------------------- Captured stdout call -----------------------------
22:41:18 32: [INFO] [launch]: All log files can be found below /home/rosbuild/.ros/log/2026-01-08-03-41-18-428632-c2c15d06c2b0-304368
22:41:18 32: [INFO] [launch]: Default logging verbosity is set to INFO
22:41:18 32: [ERROR] [launch]: Caught exception in launch (see debug for traceback): [Errno 2] No such file or directory: 'ps'
```
</details>


### Additional information

RHEL 10 support is a WIP: https://github.com/ros2/ci/pull/833

The use of `ps` in `lttng_impl.is_session_daemon_unreachable()` is a quick solution for a very rare corner case; I'm open to suggestions/alternatives if we prefer not to have to install `ps` on RHEL 10.

--- a/tracetools_trace/package.xml
+++ b/tracetools_trace/package.xml
@@ -13,6 +13,7 @@
   <author email="fixed-term.christophe.bourquebedard@de.bosch.com">Christophe Bedard</author>
 
   <exec_depend>lttngpy</exec_depend>
+  <exec_depend>procps</exec_depend>
 
   <test_depend>ament_copyright</test_depend>
   <test_depend>ament_flake8</test_depend>

